### PR TITLE
Schema.h Changes to support PAC functionality.

### PR DESCRIPTION
**Pull request by vijaya-ops**
#### Why I did it

Added below tables to support PAC functionality.

#define APP_PAC_PORT_TABLE_NAME "PAC_PORT_TABLE"
#define CFG_PAC_PORT_CONFIG_TABLE   "PAC_PORT_CONFIG_TABLE"
#define CFG_PAC_GLOBAL_CONFIG_TABLE    "PAC_GLOBAL_CONFIG_TABLE"
#define CFG_PAC_HOSTAPD_GLOBAL_CONFIG_TABLE    "HOSTAPD_GLOBAL_CONFIG_TABLE"
#define STATE_PAC_GLOBAL_OPER_TABLE               "PAC_GLOBAL_OPER_TABLE"
#define STATE_PAC_PORT_OPER_TABLE                 "PAC_PORT_OPER_TABLE"
#define STATE_PAC_AUTHENTICATED_CLIENT_OPER_TABLE "PAC_AUTHENTICATED_CLIENT_OPER_TABLE"
#define STATE_OPER_VLAN_TABLE_NAME             "OPER_VLAN"
#define STATE_OPER_VLAN_MEMBER_TABLE_NAME      "OPER_VLAN_MEMBER"
#define STATE_OPER_FDB_TABLE_NAME              "OPER_FDB"
#define STATE_OPER_PORT_TABLE_NAME             "OPER_PORT"

### CODE REVIEW

**Comment by adyeung** [comment]
@jeff-yin @ridahanif96 pls help review

**Review by ridahanif96 — COMMENTED** [review]
Changes looks good. Please re-push these changes or make a latest push to rerun pipelines to pass swss checkes.

**Review by venkatmahalingam — APPROVED** [review]

**Comment by vijaya-ops** [comment]
I see below error, Which is not related to my changes,  Any help on this is highly appreciated.

Making all in orchagent
make[3]: Entering directory '/__w/1/s/orchagent'
make[4]: Entering directory '/__w/1/s/orchagent'
g++ -DHAVE_CONFIG_H -I. -I.. -I ../lib -I .. -I ../warmrestart -I switch -I flex_counter -I debug_counter -I port -I pbh -I nhg -g -DNDEBUG  -std=c++14 -Wall -fPIC -Wno-write-strings -I/usr/include/swss -I/usr/include/libnl3 -Werror -Wno-reorder -Wcast-align -Wcast-qual -Wconversion -Wdisabled-optimization -Wextra -Wfloat-equal -Wformat=2 -Wformat-nonliteral -Wformat-security -Wformat-y2k -Wimport -Winit-self -Winvalid-pch -Wlong-long -Wmissing-field-initializers -Wmissing-format-attribute -Wno-aggregate-return -Wno-padded -Wno-switch-enum -Wno-unused-parameter -Wpacked -Wpointer-arith -Wredundant-decls -Wstack-protector -Wstrict-aliasing=3 -Wswitch -Wswitch-default -Wunreachable-code -Wunused -Wvariadic-macros -Wno-switch-default -Wno-long-long -Wno-redundant-decls -I /usr/include/sai  -Wdate-time -D_FORTIFY_SOURCE=2  -g -O2 -ffile-prefix-map=/__w/1/s=. -fstack-protector-strong -Wformat -Werror=format-security -c -o orchagent-main.o `test -f 'main.cpp' || echo './'`main.cpp
g++ -DHAVE_CONFIG_H -I. -I.. -I ../lib -I .. -I ../warmrestart -I switch -I flex_counter -I debug_counter -I port -I pbh -I nhg -g -DNDEBUG  -std=c++14 -Wall -fPIC -Wno-write-strings -I/usr/include/swss -I/usr/include/libnl3 -Werror -Wno-reorder -Wcast-align -Wcast-qual -Wconversion -Wdisabled-optimization -Wextra -Wfloat-equal -Wformat=2 -Wformat-nonliteral -Wformat-security -Wformat-y2k -Wimport -Winit-self -Winvalid-pch -Wlong-long -Wmissing-field-initializers -Wmissing-format-attribute -Wno-aggregate-return -Wno-padded -Wno-switch-enum -Wno-unused-parameter -Wpacked -Wpointer-arith -Wredundant-decls -Wstack-protector -Wstrict-aliasing=3 -Wswitch -Wswitch-default -Wunreachable-code -Wunused -Wvariadic-macros -Wno-switch-default -Wno-long-long -Wno-redundant-decls -I /usr/include/sai  -Wdate-time -D_FORTIFY_SOURCE=2  -g -O2 -ffile-prefix-map=/__w/1/s=. -fstack-protector-strong -Wformat -Werror=format-security -c -o orchagent-gearboxutils.o `test -f '../lib/gearboxutils.cpp' || echo './'`../lib/gearboxutils.cpp
g++ -DHAVE_CONFIG_H -I. -I.. -I ../lib -I .. -I ../warmrestart -I switch -I flex_counter -I debug_counter -I port -I pbh -I nhg -g -DNDEBUG  -std=c++14 -Wall -fPIC -Wno-write-strings -I/usr/include/swss -I/usr/include/libnl3 -Werror -Wno-reorder -Wcast-align -Wcast-qual -Wconversion -Wdisabled-optimization -Wextra -Wfloat-equal -Wformat=2 -Wformat-nonliteral -Wformat-security -Wformat-y2k -Wimport -Winit-self -Winvalid-pch -Wlong-long -Wmissing-field-initializers -Wmissing-format-attribute -Wno-aggregate-return -Wno-padded -Wno-switch-enum -Wno-unused-parameter -Wpacked -Wpointer-arith -Wredundant-decls -Wstack-protector -Wstrict-aliasing=3 -Wswitch -Wswitch-default -Wunreachable-code -Wunused -Wvariadic-macros -Wno-switch-default -Wno-long-long -Wno-redundant-decls -I /usr/include/sai  -Wdate-time -D_FORTIFY_SOURCE=2  -g -O2 -ffile-prefix-map=/__w/1/s=. -fstack-protector-strong -Wformat -Werror=format-security -c -o orchagent-subintf.o `test -f '../lib/subintf.cpp' || echo './'`../lib/subintf.cpp
g++ -DHAVE_CONFIG_H -I. -I.. -I ../lib -I .. -I ../warmrestart -I switch -I flex_counter -I debug_counter -I port -I pbh -I nhg -g -DNDEBUG  -std=c++14 -Wall -fPIC -Wno-write-strings -I/usr/include/swss -I/usr/include/libnl3 -Werror -Wno-reorder -Wcast-align -Wcast-qual -Wconversion -Wdisabled-optimization -Wextra -Wfloat-equal -Wformat=2 -Wformat-nonliteral -Wformat-security -Wformat-y2k -Wimport -Winit-self -Winvalid-pch -Wlong-long -Wmissing-field-initializers -Wmissing-format-attribute -Wno-aggregate-return -Wno-padded -Wno-switch-enum -Wno-unused-parameter -Wpacked -Wpointer-arith -Wredundant-decls -Wstack-protector -Wstrict-aliasing=3 -Wswitch -Wswitch-default -Wunreachable-code -Wunused -Wvariadic-macros -Wno-switch-default -Wno-long-long -Wno-redundant-decls -I /usr/include/sai  -Wdate-time -D_FORTIFY_SOURCE=2  -g -O2 -ffile-prefix-map=/__w/1/s=. -fstack-protector-strong -Wformat -Werror=format-security -c -o orchagent-recorder.o `test -f '../lib/recorder.cpp' || echo './'`../lib/recorder.cpp
g++ -DHAVE_CONFIG_H -I. -I.. -I ../lib -I .. -I ../warmrestart -I switch -I flex_counter -I debug_counter -I port -I pbh -I nhg -g -DNDEBUG  -std=c++14 -Wall -fPIC -Wno-write-strings -I/usr/include/swss -I/usr/include/libnl3 -Werror -Wno-reorder -Wcast-align -Wcast-qual -Wconversion -Wdisabled-optimization -Wextra -Wfloat-equal -Wformat=2 -Wformat-nonliteral -Wformat-security -Wformat-y2k -Wimport -Winit-self -Winvalid-pch -Wlong-long -Wmissing-field-initializers -Wmissing-format-attribute -Wno-aggregate-return -Wno-padded -Wno-switch-enum -Wno-unused-parameter -Wpacked -Wpointer-arith -Wredundant-decls -Wstack-protector -Wstrict-aliasing=3 -Wswitch -Wswitch-default -Wunreachable-code -Wunused -Wvariadic-macros -Wno-switch-default -Wno-long-long -Wno-redundant-decls -I /usr/include/sai  -Wdate-time -D_FORTIFY_SOURCE=2  -g -O2 -ffile-prefix-map=/__w/1/s=. -fstack-protector-strong -Wformat -Werror=format-security -c -o orchagent-orchdaemon.o `test -f 'orchdaemon.cpp' || echo './'`orchdaemon.cpp
orchdaemon.cpp: In member function 'virtual bool OrchDaemon::init()':
orchdaemon.cpp:122:73: error: 'CFG_SUPPRESS_ASIC_SDK_HEALTH_EVENT_NAME' was not declared in this scope
  122 |     TableConnector conf_suppress_asic_sdk_health_categories(m_configDb, CFG_SUPPRESS_ASIC_SDK_HEALTH_EVENT_NAME);
      |                                                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[4]: *** [Makefile:987: orchagent-orchdaemon.o] Error 1
make[4]: *** Waiting for unfinished jobs....
make[4]: Leaving directory '/__w/1/s/orchagent'

**Comment by vijaya-ops** [comment]
The below test is failing and I see it is failing for all the PRs in  this repo.

image

File: common/schema.h [hunk 1]
```diff
@@ -549,6 +556,16 @@ namespace swss {
 #define STATE_ACL_TABLE_TABLE_NAME                  "ACL_TABLE_TABLE"
 #define STATE_ACL_RULE_TABLE_NAME                   "ACL_RULE_TABLE"
 
+
+/*PAC*/
+#define STATE_PAC_GLOBAL_OPER_TABLE               "PAC_GLOBAL_OPER_TABLE"
+#define STATE_PAC_PORT_OPER_TABLE                 "PAC_PORT_OPER_TABLE"
+#define STATE_PAC_AUTHENTICATED_CLIENT_OPER_TABLE "PAC_AUTHENTICATED_CLIENT_OPER_TABLE"
+#define STATE_OPER_VLAN_TABLE_NAME             "OPER_VLAN"
+#define STATE_OPER_VLAN_MEMBER_TABLE_NAME      "OPER_VLAN_MEMBER"
```
Comment (matiAlfaro) [inline common/schema.h:565]: Wouldn't it be better to name it STATE_**PAC**_OPER_VLAN_TABLE_NAME? to allow for easier identification. same of actual name, something like **PAC**_OPER_VLAN

**Comment by vijaya-ops** [comment]
I am observing below error , which is not related to my changes. Any help is appreciated.

image

File: common/schema.h (hunk 1 above)
Comment (vijaya-ops) [reply common/schema.h:565]: #define STATE_OPER_VLAN_TABLE_NAME             "OPER_VLAN"
#define STATE_OPER_VLAN_MEMBER_TABLE_NAME      "OPER_VLAN_MEMBER"

The purpose of these two tables is to create VLANs and VLAN memberships dynamically. As of now the PAC is only  one, which is creating dynamic VLAN memberships. But in future any one other component can use these tables and create the dynamic VLANs and dynamic membership functionality. Hence the PAC key word is not present , for these tables.

Comment (matiAlfaro) [reply common/schema.h:565]: Thank you for your response, good to know

**Review by matiAlfaro — APPROVED** [review]

**Comment by adyeung** [comment]
@qiluo-msft code PR has been approved by PENS WG members, please help merge

**Review by adyeung — APPROVED** [review]

File: common/schema.h [hunk 2]
```diff
@@ -465,6 +467,11 @@ namespace swss {
 
 #define CFG_SUPPRESS_ASIC_SDK_HEALTH_EVENT_NAME     "SUPPRESS_ASIC_SDK_HEALTH_EVENT"
 
+#define CFG_PAC_PORT_CONFIG_TABLE   "PAC_PORT_CONFIG_TABLE"
```
Comment (qiluo-msft) [inline common/schema.h:470]: >"PAC_PORT_CONFIG_TABLE" [](http://example.com/codeflow?start=36&length=23)

Could you make them vertically aligned, like nearby lines?

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**